lib/hydra/derivatives/processors/active_encode.rb

```diff
@@ -0,0 +1,55 @@
+require 'active_encode'
+
+module Hydra::Derivatives::Processors
+  class ActiveEncodeError < StandardError
+    def initialize(status, source_path, errors = [])
+      msg = "ActiveEncode status was \"#{status}\" for #{source_path}"
+      msg = "#{msg}: #{errors.join(' ; ')}" unless errors.empty?
+      super(msg)
+    end
+  end
+
+  class ActiveEncode < Processor
+    class_attribute :timeout
+    attr_accessor :encode_class
```

Comment:
If you expect to initialize this object once and process multiple times, each with a new instance of `encode_class`, this mightn't be a viable approach.